common/reconciliation/invariant/historyExists.go

```diff
@@ -78,7 +80,16 @@ func (h *historyExists) Check(
 			InfoDetails:     errorDomainName.Error(),
 		}
 	}
-
+	domainName := domain.GetInfo().Name
```

Comment:
@demirkayaender I was talking to Aditi about this: 

The premise of an invariant check is like an integration-test or an integrated assertion, it's a check that some set of things that should be true are indeed so. 

This particular assertion in this function is that history exists and is correct. It doesn't look to domains in any way nor would anyone running this invariant expect it to care about domain status. 

My opinion is that - insofar as code organisation goes - a completely new invariant here makes sense: An assertion that the workflow exists for a domain that is active and not deprecated. That'd be less surprising for other engineers looking at the contents of the invariants, as well as easier to control with flipr rollout and similar.

Reply:
I have a separate refactoring ticket since the code will require considerable changes before we can separate out the checks like this one. For now how about we remove the check from history invariant and let the other executions go through? I have placed the same check for concreteExecution and openCurrentExecutions as well.